app/views/shared/_tag_cloud.erb

```diff
@@ -0,0 +1,4 @@
+<% tag_list.each do |tag| %>
+  <% badge_style = (tag == '<unset>') ? 'badge-danger' : 'badge-secondary' %>
+  <%= link_to(tag, "#{images_path}?tag_name=#{tag}", :class=>"js-card-tag badge #{badge_style}") %>
```

Comment:
Partial is one way to DIY it. I guess I forgot to mention that one thing we advocate in appfolio is to avoid too much logic in view and prefer using view model or presenters.
There is a academy session on this topic https://sites.google.com/a/appfolio.com/eng/new-hire-resources/engineering-academy/sessions/view-models-presenters

If I were to implement this, I'd probably create a `TagView` ruby object.
```ruby
class TagView
  attr_reader :tag

  def initial(tag:)
    @tag = tag
  end
  
  def badge_style
    (tag == '<unset>') ? 'badge-danger' : 'badge-secondary'
  end

  def tag_link
    "#{Rails.application.routes.url_helpers.images_path}?tag_name=#{tag}"
  end
end
```

And have `TagViewTest` to test the two methods.